cmd/notation/internal/osutil/read.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package osutil
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+)
+
+func ReadFile(path string) ([]byte, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return nil, err
+	}
+
+	reader := bufio.NewReader(file)
+	const bytes int64 = 10485760 //10MB in bytes
```

Comment:
> We should utilize LimitedReader to restrict the maximum file size, preventing users from inadvertently overwhelming their system. IMO to start with 10 MB is a good size limit for signature (usually signatures are of ~2KB).
Reference: https://gosamples.dev/io-limited-reader/"
https://github.com/notaryproject/notation/pull/888#discussion_r1536903028

If we allow users to pass their own size it defeats the purpose of setting a maximum size to avoid overwhelming our systems. I don't see any benefits in allowing users to set this size. 
Having said that I do see benefits in letting functions pass different max size as now the code is modularized and can be consumed for different use cases.

Reply:
By user, I meant caller of the function.

Reply:
implemented this change